src/casts/specify-representation.rkt

```diff
@@ -775,19 +775,30 @@ but a static single assignment is implicitly maintained.
              (sr-tagged-array-ref e COERCION-MEDIATING-TAG COERCION-REF-TAG-INDEX))
            (assign$ crcn-tag (sr-get-tag tmp-crcn COERCION-TAG-MASK))
            (op$ = crcn-tag COERCION-REF-SECOND-TAG))]
-        [(Ref-Coercion (app recur r) (app recur w))
+        [(Ref-Coercion (app recur r) (app recur w) (app recur flag))
          (begin$
            (assign$ second-tag
              (op$ + (op$ %<< ZERO-IMDT COERCION-SECOND-TAG-SHIFT)
                   COERCION-REF-SECOND-TAG))
-           (sr-alloc "ref-coercion" COERCION-MEDIATING-TAG
+           (if (cast-profiler?)
+               (sr-alloc "ref-coercion" COERCION-MEDIATING-TAG
+                     `(("tag" . ,second-tag)
+                       ("read-coercion" . ,r)
+                       ("write-coercion" . ,w)
+                       ("flag" . ,flag)))
+               (sr-alloc "ref-coercion" COERCION-MEDIATING-TAG
                      `(("tag" . ,second-tag)
                        ("read-coercion" . ,r)
-                       ("write-coercion" . ,w))))]        
+                       ("write-coercion" . ,w)))))]
         [(Ref-Coercion-Read (app recur e))
          (sr-tagged-array-ref e COERCION-MEDIATING-TAG COERCION-REF-READ-INDEX)]
         [(Ref-Coercion-Write (app recur e))
          (sr-tagged-array-ref e COERCION-MEDIATING-TAG COERCION-REF-WRITE-INDEX)]
+        [(Ref-Coercion-Ref-Huh (app recur e))
+         (begin$
+           (assign$ flag
+             (sr-tagged-array-ref e COERCION-MEDIATING-TAG COERCION-REF-FLAG-INDEX))
+           (op$ = flag COERCION-REF-REF-FLAG))]
```

Comment:
This needs to generate no-op when the profiler isn't being used.

Reply:
Done.